diff --git a/.gitignore b/.gitignore
index 4f42646..8b42cac 100644
--- a/.gitignore
+++ b/.gitignore
@@ -47,7 +47,6 @@ any_test.pb.*
 map*unittest.pb.*
 unittest*.pb.*
 cpp_test*.pb.*
-src/google/protobuf/compiler/js/well_known_types_embed.cc
 src/google/protobuf/util/**/*.pb.cc
 src/google/protobuf/util/**/*.pb.h

diff --git a/src/google/protobuf/compiler/js/embed.cc b/src/google/protobuf/compiler/js/embed.cc
index d04fea2..cfa84a9 100644
--- a/src/google/protobuf/compiler/js/embed.cc
+++ b/src/google/protobuf/compiler/js/embed.cc
@@ -98,7 +98,7 @@ static void AddFile(const char* name, std::basic_ostream<char>* out) {
 
 int main(int argc, char *argv[]) {
   std::cout << "#include "
-               "\"google/protobuf/compiler/js/well_known_types_embed.h\"\n";
+               "<google/protobuf/compiler/js/well_known_types_embed.h>\n";
   std::cout << "struct FileToc well_known_types_js[] = {\n";
 
   for (int i = 1; i < argc; i++) {
