diff -ru protobuf/src/google/protobuf/stubs/port.h protobuf2/src/google/protobuf/stubs/port.h
--- protobuf/src/google/protobuf/stubs/port.h	2017-07-10 16:21:22.010609708 -0700
+++ protobuf2/src/google/protobuf/stubs/port.h	2017-07-10 16:23:41.002834130 -0700
@@ -272,8 +272,12 @@
 # define GOOGLE_PROTOBUF_USE_UNALIGNED 0
 #else
 // x86 and x86-64 can perform unaligned loads/stores directly.
-# define GOOGLE_PROTOBUF_USE_UNALIGNED defined(_M_X64) || \
-     defined(__x86_64__) || defined(_M_IX86) || defined(__i386__)
+# if defined(_M_X64) || defined(__x86_64__) || defined(_M_IX86) || \
+     defined(__i386__)
+#  define GOOGLE_PROTOBUF_USE_UNALIGNED 1
+# else
+#  define GOOGLE_PROTOBUF_USE_UNALIGNED 0
+# endif
 #endif
 
 #if GOOGLE_PROTOBUF_USE_UNALIGNED
