diff -ru protobuf2/src/google/protobuf/io/coded_stream.h protobuf/src/google/protobuf/io/coded_stream.h
--- protobuf2/src/google/protobuf/io/coded_stream.h	2017-07-25 17:17:12.428661346 -0700
+++ protobuf/src/google/protobuf/io/coded_stream.h	2017-07-25 17:18:25.244795806 -0700
@@ -861,11 +861,11 @@
   bool IsSerializationDeterministic() const {
     return serialization_deterministic_is_overridden_ ?
         serialization_deterministic_override_ :
-        default_serialization_deterministic_;
+        default_serialization_deterministic_ != 0;
   }
 
   static bool IsDefaultSerializationDeterministic() {
-    return google::protobuf::internal::Acquire_Load(&default_serialization_deterministic_);
+    return google::protobuf::internal::Acquire_Load(&default_serialization_deterministic_) != 0;
   }
 
  private:
