diff -ru protobuf2/src/google/protobuf/compiler/cpp/cpp_file.cc protobuf/src/google/protobuf/compiler/cpp/cpp_file.cc
--- protobuf2/src/google/protobuf/compiler/cpp/cpp_file.cc	2017-07-26 16:15:25.479665770 -0700
+++ protobuf/src/google/protobuf/compiler/cpp/cpp_file.cc	2017-07-26 16:16:27.719778818 -0700
@@ -828,7 +828,7 @@
     printer->Print("static const char descriptor[] = {\n");
     printer->Indent();
 
-    if (file_data.size() > 66535) {
+    if (file_data.size() > 65535) {
       // Workaround for MSVC: "Error C1091: compiler limit: string exceeds 65535
       // bytes in length". Declare a static array of characters rather than use
       // a string literal. Only write 25 bytes per line.
