diff --git a/tools/genperf/perfect.c b/tools/genperf/perfect.c
index a9a14c0..e45f9c5 100644
--- a/tools/genperf/perfect.c
+++ b/tools/genperf/perfect.c
@@ -563,7 +563,7 @@ static int perfect(
         if (!augment(tabb, tabh, tabq, blen, scramble, smax, &tabb[i], nkeys, 
                      i+1, form))
         {
-          fprintf(stderr, "fail to map group of size %ld for tab size %ld\n", j, blen);
+          /* Do not print an error. The caller may retry with a larger table. */
           return FALSE;
         }
 
